pint_pandas/pint_array.py

```diff
@@ -538,10 +548,21 @@ def _from_sequence_of_strings(cls, scalars, dtype=None, copy=False):
 
     @classmethod
     def _from_factorized(cls, values, original):
+        from pandas._libs.lib import infer_dtype
```

Comment:
The _libs is meant for internal pandas use, can you use `pandas.api.types.infer_dtype` instead?

Reply:
That works fine.  I looked for other such transgressions, and see that my map method uses pandas._libs.lib for map_infer.  Should we raise an Issue to get an API for those functions?

Reply:
Yea I think so, at worst they'll tell us how we should be importing those functions

Reply:
Looking harder at it, I can rewrite this map method to call pandas.core.algorithms.map_array, which has the calls to the internal API.  We can take the return value from that function and finalize it as a PintArray.  I'll write some changes and push a fresh commit when I've tested it.